api/v1alpha1/clusterextension_types.go

```diff
@@ -485,15 +484,17 @@ type ClusterExtensionStatus struct {
 	//   - Reason: a string representation of the reason for the current state of the condition. Typically useful for building automation around particular Type+Reason combinations.
 	//   - Message: a human readable message that further elaborates on the state of the condition
 	//
-	// The current set of condition types are:
-	//   - "Installed", represents whether or not the package referenced in the spec.packageName field has been installed
+	// The global set of condition types are:
+	//   - "Installed", represents whether or not the a bundle has been installed for this ClusterExtension
 	//   - "Resolved", represents whether or not a bundle was found that satisfies the selection criteria outlined in the spec
-	//   - "Deprecated", represents an aggregation of the PackageDeprecated, ChannelDeprecated, and BundleDeprecated condition types.
-	//   - "PackageDeprecated", represents whether or not the package specified in the spec.packageName field has been deprecated
-	//   - "ChannelDeprecated", represents whether or not the channel specified in spec.channel has been deprecated
-	//   - "BundleDeprecated", represents whether or not the bundle installed is deprecated
 	//   - "Unpacked", represents whether or not the bundle contents have been successfully unpacked
 	//
+	// When the ClusterExtension is sourced from a catalog, the following conditions are also possible:
+	//   - "Deprecated", represents an aggregation of the PackageDeprecated, ChannelDeprecated, and BundleDeprecated condition types
+	//   - "PackageDeprecated", represents whether or not the package specified in the spec.source.catalog.packageName field has been deprecated
+	//   - "ChannelDeprecated", represents whether or not any channel specified in spec.source.catalog.channels has been deprecated
+	//   - "BundleDeprecated", represents whether or not the installed bundle is deprecated
```

Comment:
I'm pretty sure this line is not true, but want to double check. A quick glance at the code makes me think we should make the following edit as well:
```suggestion
	//   - "BundleDeprecated", represents whether or not the resolved bundle is deprecated
```

However, I think the original sentiment of having this condition be based on the installed bundle is what we actually want. That might require a bit more conversation and consideration though, as it isn't clear to me how exactly we would assess the deprecation status of an installed bundle during subsequent reconciles after it is installed.

Reply:
I guess we'd have two possibilities: 
 1. You want to install a bundle from the catalog, and it's deprecated -  which is straightforward
 2. You install a bundle that isn't deprecated, and in the next catalog version it is (I guess this is only meaningful for version ranges that don't let the bundle upgrade to the next (non-deprecated) version.
 
 In the 2nd case, shouldn't it pick up the condition during reconcile? Maybe it's OK that we only express deprecated conditions for resolved bundles (and not worry about the the installed bundles) since either the CE will get upgraded, or if it can't/won't, we'd anyway resolve to the installed bundle?

Reply:
My initial thought is that we only touch the `BundleDeprecated` condition when:
1. We successfully install a bundle. In this case the resolved bundle and the installed bundle are the same, so the resolved deprecation is applicable to the installed bundle.
2. We detect that a bundle is somehow no longer installed, in which case we unset (or set to `Unknown`) the `BundleDeprecated` condition.